### PR TITLE
Add parsing support for LATERAL keyword

### LINKED ISSUE #9716: parser: support LATERAL keyword
### Feature Description

We have some high throughput queries that really benefit from the semi-recent `LATERAL` keyword. Right now, Vitess just returns `[HY000][1105] syntax error at position ...` for queries that run on >= MySQL 8.0.14.
https://dev.mysql.com/doc/refman/8.0/en/lateral-derived-tables.html

### Use Case(s)

Finding the most recent value per group by

```sql
SELECT
  salesperson.name,
  max_sale.amount,
  max_sale.customer_name
FROM
  salesperson,
  -- find maximum size and customer at same time
  LATERAL
  (SELECT amount, customer_name
    FROM all_sales
    WHERE all_sales.salesperson_id = salesperson.id
    ORDER BY amount DESC LIMIT 1)
  AS max_sale;
```

### PR DESCRIPTION
**Pull request by K-Kumar-01**
## Description

Adds parsing support for LATERAL keyword. More information [here](https://dev.mysql.com/doc/refman/8.0/en/lateral-derived-tables.html)

## Related Issue(s)

Fixes #9716
#8604

## Checklist
- [x] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/sqlparser/ast_format.go [hunk 1]
```diff
@@ -1199,6 +1199,9 @@ func (node *Subquery) Format(buf *TrackedBuffer) {
 
 // Format formats the node.
 func (node *DerivedTable) Format(buf *TrackedBuffer) {
+	if node.LateralPresent {
+		buf.WriteString("lateral")
```
Comment (GuptaManan100) [inline go/vt/sqlparser/ast_format.go:1203]: We generally have a space between a keyword and opening parenthesis . Could you add a space here too

**Review by GuptaManan100 — COMMENTED** [review]
Other than this all the changes look good 🥳

File: go/vt/sqlparser/ast_format.go (hunk 1 above)
Comment (K-Kumar-01) [reply go/vt/sqlparser/ast_format.go]: Done

File: go/vt/sqlparser/ast_format.go [hunk 2]
```diff
@@ -1199,6 +1199,9 @@ func (node *Subquery) Format(buf *TrackedBuffer) {
 
 // Format formats the node.
 func (node *DerivedTable) Format(buf *TrackedBuffer) {
+	if node.Lateral {
+		buf.WriteString(" lateral")
```
Comment (systay) [inline go/vt/sqlparser/ast_format.go:1203]: are you sure about this space? I added an couple of examples to parse test, and they are failing:
```
	{
		input:  "select * from lateral (select 1) as x(user)",
		output: "select * from lateral (select 1 from dual) as x(`user`)",
	}, {
		input:  "select * from apa, lateral (select 1) as x(user)",
		output: "select * from apa, lateral (select 1 from dual) as x(`user`)",
	}
```

Output:
```
--- FAIL: TestValid (0.03s)
    --- FAIL: TestValid/select_*_from_lateral_(select_1)_as_x(user) (0.00s)
        parse_test.go:2318: Parsing failed.
            Expected/Got:
            select * from lateral (select 1 from dual) as x(`user`)
            select * from  lateral(select 1 from dual) as x(`user`)
    --- FAIL: TestValid/select_*_from_apa,_lateral_(select_1)_as_x(user) (0.00s)
        parse_test.go:2318: Parsing failed.
            Expected/Got:
            select * from apa, lateral (select 1 from dual) as x(`user`)
            select * from apa,  lateral(select 1 from dual) as x(`user`)
```

Comment (GuptaManan100) [reply go/vt/sqlparser/ast_format.go:1203]: Yeah, the space is on the wrong side of lateral! 😆

Comment (K-Kumar-01) [reply go/vt/sqlparser/ast_format.go:1203]: My bad, I have fixed this.

Comment (GuptaManan100) [reply go/vt/sqlparser/ast_format.go]: You should also add the tests that @systay ran into the parse_test file

**Review by GuptaManan100 — APPROVED** [review]

**Review by systay — APPROVED** [review]

**Pull request merged**